### PR TITLE
[PL-52014] Update delegate docs on number of parallel executions

### PR DESCRIPTION
**Pull request by brian-r-fisher**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: Update delegate docs on the number of parallel executions
* Jira/GitHub Issue numbers (if any): PL-52014
* Preview links/images (Internal contributors only): 
   * https://667b1819d506d3273137e8d9--harness-developer.netlify.app/docs/faqs/harness-delegate-faqs#what-are-the-delegate-system-requirements
   * https://667b1819d506d3273137e8d9--harness-developer.netlify.app/docs/platform/delegates/delegate-concepts/delegate-overview#delegate-sizes
   * https://667b1819d506d3273137e8d9--harness-developer.netlify.app/docs/platform/delegates/delegate-concepts/delegate-requirements

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [x] Successful preview build.
- [x] Code owner review.
- [x] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: docs/faqs/harness-delegate-faqs.md [hunk 1]
```diff
@@ -149,6 +149,12 @@ For more information, go to [Automate delegate installation](../platform/delegat
 
 ### What are the delegate system requirements?
 
+:::danger delegate resources
+ Remember that the memory and CPU requirements are for the delegate only. Your delegate host/pod/container will need more computing resources for its operations systems and other services, such as Docker or Kubernetes.
+
+ The resource requirements of the delegate container vary depending on the type of tasks or executions. For instance, CI-only delegates can handle hundreds of parallel pipelines, whereas for CD Terraform tasks, a single task might require a 2Gi container.
```
Comment (brian-r-fisher) [inline docs/faqs/harness-delegate-faqs.md:155]:
```suggestion
 The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```

File: docs/platform/delegates/delegate-concepts/delegate-overview.md [hunk 2]
```diff
@@ -58,6 +58,12 @@ For advanced installation topics, go to the following:
 
 ### Delegate sizes
 
+:::danger delegate resources
+ Remember that the memory and CPU requirements are for the delegate only. Your delegate host/pod/container will need more computing resources for its operations systems and other services, such as Docker or Kubernetes.
```
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-overview.md:62]:
```suggestion
Memory and CPU requirements are for the delegate only. Your delegate host/pod/container requires additional computing resources for its operating system and other services, such as Docker or Kubernetes.
```

File: docs/platform/delegates/delegate-concepts/delegate-overview.md [hunk 3]
```diff
@@ -58,6 +58,12 @@ For advanced installation topics, go to the following:
...
+
+ The resource requirements of the delegate container vary depending on the type of tasks or executions. For instance, CI-only delegates can handle hundreds of parallel pipelines, whereas for CD Terraform tasks, a single task might require a 2Gi container.
```
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-overview.md:64]:
```suggestion
The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```

File: docs/platform/delegates/delegate-concepts/delegate-requirements.md [hunk 4]
```diff
@@ -24,15 +24,20 @@ Note the following important information about delegates:
 
 * The delegate is installed in your network and connects to the Harness Manager.
 
-  The requirements for memory and CPU support only the delegate. The delegate host, pod, and container require additional compute resources to support other operations, systems, and services, such as Docker or Kubernetes.
+ :::danger delegate resources
+ Remember that the memory and CPU requirements are for the delegate only. Your delegate host/pod/container will need more computing resources for its operations systems and other services, such as Docker or Kubernetes.
```
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-requirements.md:28]:
```suggestion
Memory and CPU requirements are for the delegate only. Your delegate host/pod/container requires additional computing resources for its operating system and other services, such as Docker or Kubernetes.
```

File: docs/platform/delegates/delegate-concepts/delegate-requirements.md [hunk 5]
```diff
@@ -24,15 +24,20 @@ Note the following important information about delegates:
...
 
-  | Replicas | Required memory / CPU | Maximum parallel deployments and builds across replicas |
-  | :--: |  :--: | :--: |
-  | 1 | 2 GB / 0.5 CPU | 10 |
-  | 2 | 4 GB / 1 CPU | 20 |
-  | 4 | 8 GB / 2 CPU | 40 |
-  | 8 | 16 GB / 4 CPU | 80 |
+ The resource requirements of the delegate container vary depending on the type of tasks or executions. For instance, CI-only delegates can handle hundreds of parallel pipelines, whereas for CD Terraform tasks, a single task might require a 2Gi container.
```
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-requirements.md:30]:
```suggestion
The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```

File: docs/platform/delegates/delegate-concepts/delegate-requirements.md [hunk 6]
```diff
@@ -24,15 +24,20 @@ Note the following important information about delegates:
...
+ :::
 
+One delegate size does not fit all use cases, so Harness lets you pick from several options:
+
+| Replicas | Required memory / CPU | Maximum parallel deployments and builds across replicas |
+| :--: |  :--: | :--: |
+| 1 | 2 GB / 0.5 CPU | 10 |
+| 2 | 4 GB / 1 CPU | 20 |
+| 4 | 8 GB / 2 CPU | 40 |
+| 8 | 16 GB / 4 CPU | 80 |
```
Comment (mdmilic) [inline docs/platform/delegates/delegate-concepts/delegate-requirements.md:40]: I think we used to have this in our old installation flows as options, but those don't apply any more. I think we should just remove this.

File: docs/platform/delegates/delegate-concepts/delegate-overview.md [hunk 7]
```diff
@@ -58,6 +58,12 @@ For advanced installation topics, go to the following:
 
 ### Delegate sizes
 
+:::danger delegate resources
+Memory and CPU requirements are for the delegate only. Your delegate host/pod/container requires additional computing resources for its operating system and other services, such as Docker or Kubernetes.
+
+The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```
Comment (mdmilic) [inline docs/platform/delegates/delegate-concepts/delegate-overview.md:64]: Maybe just clarify here that this is due to terraform memory requirements, so it's clear that it's not delegate having an issue. E.g. something like 
a single task might require a 2Gi container due to terraform memory requirements. 

Also, I read in some Hashicorp docs that they recomend 500Mb min per terraform command. Not sure if we should include this somehow in the doc. Linking their doc wouldn't be good as it points to their own runners (delegate equivalent).

File: docs/platform/delegates/delegate-concepts/delegate-requirements.md (hunk 6 above)
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-requirements.md:40]:
```suggestion
```

File: docs/faqs/harness-delegate-faqs.md [hunk 8]
```diff
@@ -149,6 +149,12 @@ For more information, go to [Automate delegate installation](../platform/delegat
 
 ### What are the delegate system requirements?
 
+:::danger delegate resources
+Memory and CPU requirements are for the delegate only. Your delegate host/pod/container requires additional computing resources for its operating system and other services, such as Docker or Kubernetes.
+
+The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```
Comment (brian-r-fisher) [inline docs/faqs/harness-delegate-faqs.md:155]:
```suggestion
The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container due to Terraform memory requirements. Each Terraform command requires a minimum of 500Mb of memory. 
```

File: docs/platform/delegates/delegate-concepts/delegate-overview.md (hunk 7 above)
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-overview.md:64]:
```suggestion
The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container due to Terraform memory requirements. Each Terraform command requires a minimum of 500Mb of memory. 
```

File: docs/platform/delegates/delegate-concepts/delegate-requirements.md [hunk 9]
```diff
@@ -24,15 +24,20 @@ Note the following important information about delegates:
...
+ :::danger delegate resources
+Memory and CPU requirements are for the delegate only. Your delegate host/pod/container requires additional computing resources for its operating system and other services, such as Docker or Kubernetes.
 
-  | Replicas | Required memory / CPU | Maximum parallel deployments and builds across replicas |
-  | :--: |  :--: | :--: |
-  | 1 | 2 GB / 0.5 CPU | 10 |
-  | 2 | 4 GB / 1 CPU | 20 |
-  | 4 | 8 GB / 2 CPU | 40 |
-  | 8 | 16 GB / 4 CPU | 80 |
+The resource requirements of the delegate container vary depending on the type of tasks or executions. For example, CI-only delegates can handle hundreds of parallel pipelines, whereas, for CD Terraform tasks, a single task might require a 2Gi container.
```
Comment (brian-r-fisher) [inline docs/platform/delegates/delegate-concepts/delegate-requirements.md:30]:
```suggestion
The resource requirements for the delegate container depend on the type of tasks or executions. For instance, CI-only delegates can handle hundreds of parallel pipelines. However, for CD Terraform tasks, a single task might require a 2Gi container due to Terraform's memory requirements. Each Terraform command needs at least 500MB of memory.
```

**Review by mdmilic — APPROVED** [review]

**Pull request merged**